### PR TITLE
feat: example agent using IBM Granite LLM

### PR DESCRIPTION
**Pull request by grahamwhiteuk**
### Which issue(s) does this pull-request address?

Contributes-to: https://github.com/i-am-bee/internal/issues/19

### Description

Initial implementation of a Bee Agent using the IBM Granite Model.

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [x] Unit tests pass: `yarn test:unit`
- [x] E2E tests pass: `yarn test:e2e`
- [ ] Tests are included 
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: examples/agents/ibmGraniteBee.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  if (!provider) {
+    provider = parseEnv("LLM_BACKEND", z.nativeEnum(Providers), Providers.OLLAMA);
+  }
+
+  const factory = LLMFactories[provider];
+  if (!factory) {
+    throw new Error(`Provider "${provider}" not found.`);
+  }
+  return factory();
+}
```
Comment (Tomas2D) [inline examples/agents/ibmGraniteBee.ts:84]: Could you put everything into one function to make the example more readable?

File: examples/agents/ibmGraniteBee.ts
Comment (Tomas2D) [inline examples/agents/ibmGraniteBee.ts:1]: Rename to `ibm_granite_bee.ts` to preserve consistency.

**Review by Tomas2D — CHANGES_REQUESTED** [review]

File: examples/agents/ibmGraniteBee.ts (hunk 1 above)
Comment (grahamwhiteuk) [reply examples/agents/ibmGraniteBee.ts]: Resolved in 1e88fe1

File: examples/agents/ibmGraniteBee.ts
Comment (grahamwhiteuk) [reply examples/agents/ibmGraniteBee.ts:1]: Resolved in de657cd

**Review by Tomas2D — APPROVED** [review]

File: examples/agents/ibm_granite_bee.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+function getChatLLM(provider?: Provider): ChatLLM<ChatLLMOutput> {
+  const LLMFactories: Record<Provider, () => ChatLLM<ChatLLMOutput>> = {
+    [Providers.OPENAI]: () =>
+      new OpenAIChatLLM({
+        modelId: getEnv("OPENAI_MODEL") || "ibm-granite/granite-3-8b-instruct",
+        parameters: {
+          temperature: 0,
+          max_tokens: 2048,
+        },
+      }),
```
Comment (Tomas2D) [inline examples/agents/ibm_granite_bee.ts:38]: Granite is not available in OpenAI directly. Remove.

Comment (grahamwhiteuk) [reply examples/agents/ibm_granite_bee.ts]: Resolved in 213f963

**Pull request merged**